### DISCUSSION TITLE
nvm not recognized

### DISCUSSION THREAD

**Discussion by sdarwin** [coreybutler/nvm-windows#1166] [Support] [answered]
Installing nvm-windows with chocolatey.  "choco install nvm".   
After installing, it says

```
nvm : The term 'nvm' is not recognized as the name of a cmdlet, function, script file, or operable program.
```

If I close and open the powershell window, it's fixed.

However this is part of a larger automation script I would prefer to be able to use the command ASAP.  Could I manually refresh the necessary variables?  Is it only PATH?  I see there are NVM_HOME and NVM_SYMLINK and PATH.

Here is an old conversation from the choco discussion forums.

```
Installed on my win 10, and when i type "nvm", this is what i see - "'nvm' is not recognized as an internal or external command,
operable program or batch file."

You may need to open a new command prompt or run `refreshenv` to make it show up. A reboot will fix it as well.

refreshenv worked for me! Thanks!

Had to open a new command prompt to get the nvm -v to work. refreshenv didn't work for me though
```

Usually, I would run `refreshenv`.  With other chocolatey installations that is effective.  But not for `nvm`.

What commands could I run manually to refresh NVM_HOME, NVM_SYMLINK, PATH, and not be forced to open a new terminal window?

**Comment by coreybutler** [comment, marked as answer]
We do not maintain the choco installer. I don't know what's going on with it, but if it wraps the setup.exe we provide, there is likely a race condition that isn't accounted for. The InnoSetup script attempts to refresh the environment, but Windows doesn't provide much in the way of alerting when the install process is complete. It's an annoyance I've never been able to address in the InnoSetup scripts. That said, this is just a guess a what may be going on.

There's a detailed guide showing what the official installer does though, available a https://github.com/coreybutler/nvm-windows/wiki#manual-installation. I think it provides what you're looking for.

**Comment by sdarwin** [comment]
Thanks for the info. Running `nvm-setup.exe` required closing/opening the terminal.  That `#manual-installation` looks like it contains the steps.
